File: src/main/java/com/ghyinc/finance/global/init/DataInitializer.java
package com.ghyinc.finance.global.init;

import com.ghyinc.finance.domain.loan.entity.Partner;
import com.ghyinc.finance.domain.loan.entity.PartnerLoanType;
import com.ghyinc.finance.domain.loan.entity.Product;
import com.ghyinc.finance.domain.loan.enums.LoanType;
import com.ghyinc.finance.domain.loan.enums.PartnerCode;
import com.ghyinc.finance.domain.loan.enums.PartnerType;
import com.ghyinc.finance.domain.loan.repository.PartnerLoanTypeRepository;
import com.ghyinc.finance.domain.loan.repository.PartnerRepository;
import com.ghyinc.finance.domain.loan.repository.ProductRepository;
import com.ghyinc.finance.domain.user.entity.Member;
import com.ghyinc.finance.domain.user.repository.MemberRepository;
import com.ghyinc.finance.global.crypto.enums.CryptoAlgorithm;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Objects;

/**
 * 서버 기동 시 Partner 테이블 초기 데이터 Insert
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class DataInitializer implements ApplicationRunner {
    private final PartnerRepository partnerRepository;
    private final PartnerLoanTypeRepository partnerLoanTypeRepository;
    private final ProductRepository productRepository;
    private final MemberRepository memberRepository;

    @Override
    public void run(ApplicationArguments args) throws Exception {
        List<Partner> initialPartner = List.of(
                Partner.builder()
                        .partnerCode(PartnerCode.KAKAO_BANK)
                        .partnerName(PartnerCode.KAKAO_BANK.getPartnerName())
                        .partnerType(PartnerType.BANK)
                        .active(true)
                        .algorithm(CryptoAlgorithm.AES_256_CBC)
                        .cryptoKey("wvtX75QJj1Uw1xKqw2kyPOVNBAmDr2vr")
                        .build(),
                Partner.builder()
                        .partnerCode(PartnerCode.TOSS_BANK)
                        .partnerName(PartnerCode.TOSS_BANK.getPartnerName())
                        .partnerType(PartnerType.BANK)
                        .active(true)
                        .algorithm(CryptoAlgorithm.AES_256_CBC)
                        .cryptoKey("bd0001eb9404dc257b90547d1343c4de")
                        .build(),
                Partner.builder()
                        .partnerCode(PartnerCode.KB_CAPITAL)
                        .partnerName(PartnerCode.KB_CAPITAL.getPartnerName())
                        .partnerType(PartnerType.CAPITAL)
                        .active(true)
                        .algorithm(CryptoAlgorithm.RSA_OAEP)
                        .publicKey("MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAyxkSWuJRXR27YstInQ+0\n" +
                                "Dohwf82TcqvYtCeW1My8wNoBVhArdT9SBGdcU3z4RYEqioRPm4MCoaJBwAlL/wv6\n" +
                                "Tp3+3ZkDlINrv/cTOxfvRBSB4rd6EVHNT53oXUx7mRFPv0/uOdCyKlELtTcCRriO\n" +
                                "R9mPHY/b99clKS/NYPvPgNa+H5dhnX08wa4wm3n+N+uAjVtppKWdq+aGPBFyU50q\n" +
                                "xmVvQV9JcdX25CyNB9djtWt1EfJ2qq1NqTt06ciTwbu3pyDuRLByjB5HtusSRJrj\n" +
                                "kKZ0MWPgSgfDcvXk4GzA9UBpnYc25cl0L3JTtjeZDSnX4lgeoc7x8x6sxsnvDsXt\n" +
                                "KQIDAQAB")
                        .privateKey("MIIEpAIBAAKCAQEAyxkSWuJRXR27YstInQ+0Dohwf82TcqvYtCeW1My8wNoBVhAr\n" +
                                "dT9SBGdcU3z4RYEqioRPm4MCoaJBwAlL/wv6Tp3+3ZkDlINrv/cTOxfvRBSB4rd6\n" +
                                "EVHNT53oXUx7mRFPv0/uOdCyKlELtTcCRriOR9mPHY/b99clKS/NYPvPgNa+H5dh\n" +
                                "nX08wa4wm3n+N+uAjVtppKWdq+aGPBFyU50qxmVvQV9JcdX25CyNB9djtWt1EfJ2\n" +
                                "qq1NqTt06ciTwbu3pyDuRLByjB5HtusSRJrjkKZ0MWPgSgfDcvXk4GzA9UBpnYc2\n" +
                                "5cl0L3JTtjeZDSnX4lgeoc7x8x6sxsnvDsXtKQIDAQABAoIBAQCi0T+owoSNzLcb\n" +
                                "lXJqD1u+xtzBaFILfP6mNpKxmEy9oket8hqUzSV4SFB40dfLCKjNERMszZN/dq+V\n" +
                                "Px7AoZ6SBhF7Hx8CoXTxGSc+mYqEHpid448lcVnRuPq+SQFRDdLLwU1u5gLe78ge\n" +
                                "B7J4dZ4CtcQI4/ppLv4ojZztYhHQ62k4Gwwy1lqlfVM08+jVcJlpgDOjt+KZpExA\n" +
                                "Uf4cUoMOg7cx0o8M1ROZYWxaPBXFNF9v4QsRZcpeKwqIqlpWsHch89NAx1oMUVVx\n" +
                                "K1nl4Xj2KtMZw32Af1eGAQldSO3qCy70PiEeqOAQE/j6TZPxw5upX1KA7o348yB5\n" +
                                "yACd+J8BAoGBAOSihM5r3OZcUtx3zPAE5U/+4VabzflXEs0Owi0nquyCgy0wdlPs\n" +
                                "PzYJOk2KSWqkoUOnEOZlAxLOoNz2haiA1FV6EriHXmhs87IvVCuFGkpzZ1FDRGCF\n" +
                                "QWz5IbJPbPt3R7IOGE2+guDZt4TAcNjthXBN3AUOj+r18Jka2aqlZMyJAoGBAONo\n" +
                                "FwgqVFKxi15rTB8f+uVRlhJLVzRLNEdbCFzJmGLkZGzjcfsziz4Ch+uJp+oZuvQQ\n" +
                                "1CtLsqkcU/eTbGEwZENz0LAWO3D/eSkMlRkiWbI+bM0GHNocbZ4wTw6/HxYkm64f\n" +
                                "+kd1S5m6WR90HpKO+42nhaBap+es5iaXn1n0EjOhAoGANHDydUZYTJ4wg1EXOJZm\n" +
                                "4opbtTnXbLGEJnSUJTdMBSOKYvsSqP0vIn3LWa22WTeZpaLURYQ1yEKMsyH4VkX2\n" +
                                "bgSp9plWFi2nV99zNug4t4rwz7rWHC10bEJYcEW3gZZCY5zIBk0ER/6oEVLyj08r\n" +
                                "pC63oJFOgV4X6YY3FuUI0cECgYEAljnHLU+5UL+VEBTVvqIDvsX8260FuLgNmy3a\n" +
                                "AmHy1zGF3iEKxSWx0I8fd0wCrzW8OUt8vfVN20Wpep3bNQEg2yaBMDIfpnA+fA2h\n" +
                                "2W7FzmhKu85T9Qpep+fF8jnzsU8RwR/C2L316WIfShYNtEfciiGmtt3smbGwgMId\n" +
                                "NPF1rMECgYAoKyRsqnw6HMGvWQanBn7pHhy2Kh3bxdwu/0j+1+5hF17FmGVnvMxE\n" +
                                "heJWnTHr2tnB4QnauZsCafZmNzl8u52Nds4QKlBLFRLhmQEmsO1EpkE8ZLJu+jwQ\n" +
                                "gaWmBN05DhJC2NpwBJ/o7lWvGRUgsT6X4LWJaPgVfWpAjjUDT44x5g==")
                        .build(),
                Partner.builder()
                        .partnerCode(PartnerCode.K_BANK)
                        .partnerName(PartnerCode.K_BANK.getPartnerName())
                        .partnerType(PartnerType.BANK)
                        .active(true)
                        .algorithm(CryptoAlgorithm.AES_256_CBC)
                        .cryptoKey("AAEGRJuHwTWvrYsaa0V7vAqk+wZuSa2l")
                        .build(),
                Partner.builder()
                        .partnerCode(PartnerCode.LINE_BANK)
                        .partnerName(PartnerCode.LINE_BANK.getPartnerName())
                        .partnerType(PartnerType.BANK)
                        .active(true)
                        .algorithm(CryptoAlgorithm.AES_256_CBC)
                        .cryptoKey("AAEGRJuHwTWvrYsaa0V7vAqk+wZuSa2l")
                        .build(),
                Partner.builder()
                        .partnerCode(PartnerCode.SHINHAN_BANK)
                        .partnerName(PartnerCode.SHINHAN_BANK.getPartnerName())
                        .partnerType(PartnerType.BANK)
                        .active(false)
                        .algorithm(CryptoAlgorithm.AES_256_CBC)
                        .cryptoKey("AAEGRJuHwTWvrYsaa0V7vAqk+wZuSa2l")
                        .build()
        );
        partnerRepository.saveAll(initialPartner);

        List<PartnerLoanType> initialPartnerLoanType = List.of(
                PartnerLoanType.create(
                            initialPartner.stream()
                                    .filter(partner -> Objects.equals(PartnerCode.KAKAO_BANK, partner.getPartnerCode()))
                                    .findFirst()
                                    .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   true
                ),
                PartnerLoanType.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.TOSS_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   true
                ),
                PartnerLoanType.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.LINE_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   true
                ),
                PartnerLoanType.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.KB_CAPITAL, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.BUSINESS
                        ,   true
                ),
                PartnerLoanType.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.SHINHAN_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   true
                )
        );

        partnerLoanTypeRepository.saveAll(initialPartnerLoanType);

        List<Product> initialProduct = List.of(
                Product.create(
                           initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.LINE_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   "P060100206"
                        ,   "사잇돌"
                        ,   true
                        ),
                Product.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.LINE_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   "P060100205"
                        ,   "드림론"
                        ,   true
                ),
                Product.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.KAKAO_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   "TA"
                        ,   "갈아타기OK론"
                        ,   true
                ),
                Product.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.TOSS_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   "FNQ005"
                        ,   "kiwi비상금"
                        ,   true
                ),
                Product.create(
                        initialPartner.stream()
                                .filter(partner -> Objects.equals(PartnerCode.SHINHAN_BANK, partner.getPartnerCode()))
                                .findFirst()
                                .orElse(null)
                        ,   LoanType.PERSONAL_CREDIT
                        ,   "0201001074"
                        ,   "비상금신한론"
                        ,   true)
        );
        productRepository.saveAll(initialProduct);

        List<Member> initialUser = List.of(
                Member.builder()
                        .name("윤교희")
                        .mobile("01056677055")
                        .email("gyohee91@gmail.com")
                        .build()
        );

        memberRepository.saveAll(initialUser);
    }
}

File: src/main/java/com/ghyinc/finance/global/init/DataInitializer.kt
package com.ghyinc.finance.global.init

import com.ghyinc.finance.domain.loan.entity.Partner
import com.ghyinc.finance.domain.loan.entity.PartnerLoanType
import com.ghyinc.finance.domain.loan.entity.Product
import com.ghyinc.finance.domain.loan.enums.LoanType
import com.ghyinc.finance.domain.loan.enums.PartnerCode
import com.ghyinc.finance.domain.loan.enums.PartnerType
import com.ghyinc.finance.domain.loan.repository.PartnerLoanTypeRepository
import com.ghyinc.finance.domain.loan.repository.PartnerRepository
import com.ghyinc.finance.domain.loan.repository.ProductRepository
import com.ghyinc.finance.domain.user.entity.Member
import com.ghyinc.finance.domain.user.repository.MemberRepository
import com.ghyinc.finance.global.crypto.enums.CryptoAlgorithm
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.stereotype.Component

/**
 * 서버 기동 시 Partner 테이블 초기 데이터 Insert
 */
@Component
class DataInitializer(
    private val partnerRepository: PartnerRepository,
    private val partnerLoanTypeRepository: PartnerLoanTypeRepository,
    private val productRepository: ProductRepository,
    private val memberRepository: MemberRepository
) : ApplicationRunner {

    @Throws(Exception::class)
    override fun run(args: ApplicationArguments) {
        val initialPartner = listOf(
            Partner().apply {
                partnerCode = PartnerCode.KAKAO_BANK
                partnerName = PartnerCode.KAKAO_BANK.partnerName
                partnerType = PartnerType.BANK
                active = true
                algorithm = CryptoAlgorithm.AES_256_CBC
                cryptoKey = "wvtX75QJj1Uw1xKqw2kyPOVNBAmDr2vr"
            },
            Partner().apply {
                partnerCode = PartnerCode.TOSS_BANK
                partnerName = PartnerCode.TOSS_BANK.partnerName
                partnerType = PartnerType.BANK
                active = true
                algorithm = CryptoAlgorithm.AES_256_CBC
                cryptoKey = "bd0001eb9404dc257b90547d1343c4de"
            },
            Partner().apply {
                partnerCode = PartnerCode.KB_CAPITAL
                partnerName = PartnerCode.KB_CAPITAL.partnerName
                partnerType = PartnerType.CAPITAL
                active = true
                algorithm = CryptoAlgorithm.RSA_OAEP
                publicKey = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAksA93OlyL+X08oCGVDRd36PhcC10ToB+xgcVfFbnXn21mc3hcxxVnLb509JBcwImRt3VHGTIzHiEUmzNKI3oaEP4dq3iHk6WAAy5gn76pXCPXbp+7yweRKhWLtGT9W+WVQvO7WMi6VLuQQmPvQ0+Au02/bmrnjauPS0qHftTiyjAhOy//LMAr861aXxq8PZK9zpRYiW/KHaPoPIhxChgVCBo9tdXTK7hCp0//ARaJ1WdC1M3bzRMBNmsKGphisa0eEQh1ArXoBA0tJtOKWNcVRuj8355u97a2dh2U8dFZ27E2phf97XLFlcOwtMNs4C1xXzF6kDdTJNeEn0dWSNhywIDAQAB"
                privateKey = "MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCSwD3c6XIv5fTygIZUNF3fo+FwLXROgH7GBxV8VudefbWZzeFzHFWctvnT0kFzAiZG3dUcZMjMeIRSbM0ojehoQ/h2reIeTpYADLmCfvqlcI9dun7vLB5EqFYu0ZP1b5ZVC87tYyLpUu5BCY+9DT4C7Tb9uaueNq49LSod+1OLKMCE7L/8swCvzrVpfGrw9kr3OlFiJb8odo+g8iHEKGBUIGj211dMruEKnT/8BFonVZ0LUzdvNEwE2awoamGKxrR4RCHUCtegEDS0m04pY1xVG6Pzfnm73trZ2HZTx0VnbsTamF/3tcsWVw7C0w2zgLXFfMXqQN1Mk14SfR1ZI2HLAgMBAAECggEANuLm4FqBpi+EOzTdyN2jJJN3LE0b3IaLo8Yff5dDRuCOjN1nTnXRAWGgM9zGNN0fcs7iN5HsVilXGDrwSpQ00dZmNgEX+szapl8b/RY/6aOdQ2mOsyaVlYk+XpltFNVs7RcTCE9VTd4CFVodpBAxWsfdEu2gJfb+mQvs3jTTTM7aoH/M7lUPtif7Nm4oMDiw+XBC1DBA+78vI7I8xHidxWwlNERh7eDrxnnqPoA7zkBIo1NKd3tiY3Z8Yz4r9/sqry/ZWayXVJAMYJhK1/AnbyTXsj27lG2nDUOt2d8KKE6ENMpeBOvkQYNUe0ftHQbZ4Cjds0DSenszX3xgg+vWbQKBgQDARXWeoXJ80881yz3GQXhU9Gn/0XL+DLIlcLlV519yZUzmhaKqs3ui4NP2XaH9MbgaBGQMW8QTXwXtHWNVwHNERmspCbMWYRgZXMV7vnS6+Q0ontzNiD+iXI29JKdnnghYsHfK3Z1DooxlcX5uHXR8a4Ur+qvkAfByC1m4QgonRwKBgQDDZE1oLSWO9uELgc0201ZdjV4ddHzpzxx2+83oJJAhyuZCs/KeXEVfRE1DCNn0fiJUj2xlGbIn9bBnsmYBsc1ERg/MMaank/rhXdVmTFdqxUBr+heypNqt0wOz02PbZDNMcqp06QfNEtKdV6fJE2Hce4rZPcwhu6piw8AeYsx7XQKBgE4Hom1nbhQ0zaIyQnbEOTaZHq+ga/+oXWAOeFjhln3RmLzxPQvz3VhD0CHq7APaerGsWIIfd9q4tKn/REIX4W+y+GhmNFT/wI0Cdm7641rCGlIC7u9GvTSCRU+eoYXOv+pma6db+yfovvuobLv5nj2kUR6BE+Nr2g4ehyHdQ0ufAoGALaqCoOd0UL6IiFz07mxkUgcZSP8Rtr+OIozlae0ptowqVsqh2LsuB376I8Gs1wSn12WR6usfhVFZwlKinqDEncFWLd4o2h+u0f9RQdBz2eNyFApmgX0gEuIvilbMjtkTWDmwdxSDmz1b9iQndcpO4+4H3JAh5nxq7RxdCX0D140CgYBLzEpO5AtTNiFnTvIxnwRpH0cu7zJ+Gh11gGFjFD6Pg96mBUt+CCsto6YcNzNSqczy+jD+VQ5VnTa2oYqO4R0nmLLeJTkjIiTUVo3WiP3npbjiW6wzyo0u0HwKZrO5iRnjoJQS/4NG6d0qPE2SKEs9X41MOEaJYbTuz9pMgFJymQ=="
            },
            Partner().apply {
                partnerCode = PartnerCode.K_BANK
                partnerName = PartnerCode.K_BANK.partnerName
                partnerType = PartnerType.BANK
                active = true
                algorithm = CryptoAlgorithm.AES_256_CBC
                cryptoKey = "AAEGRJuHwTWvrYsaa0V7vAqk+wZuSa2l"
            },
            Partner().apply {
                partnerCode = PartnerCode.LINE_BANK
                partnerName = PartnerCode.LINE_BANK.partnerName
                partnerType = PartnerType.BANK
                active = true
                algorithm = CryptoAlgorithm.AES_256_CBC
                cryptoKey = "AAEGRJuHwTWvrYsaa0V7vAqk+wZuSa2l"
            },
            Partner().apply {
                partnerCode = PartnerCode.SHINHAN_BANK
                partnerName = PartnerCode.SHINHAN_BANK.partnerName
                partnerType = PartnerType.BANK
                active = false
                algorithm = CryptoAlgorithm.AES_256_CBC
                cryptoKey = "AAEGRJuHwTWvrYsaa0V7vAqk+wZuSa2l"
            }
        )
        partnerRepository.saveAll(initialPartner)

        val initialPartnerLoanType = listOf(
            PartnerLoanType.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.KAKAO_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                active = true
            ),
            PartnerLoanType.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.TOSS_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                active = true
            ),
            PartnerLoanType.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.LINE_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                active = true
            ),
            PartnerLoanType.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.KB_CAPITAL },
                loanType = LoanType.BUSINESS,
                active = true
            ),
            PartnerLoanType.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.SHINHAN_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                active = true
            )
        )

        partnerLoanTypeRepository.saveAll(initialPartnerLoanType)

        val initialProduct = listOf(
            Product.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.LINE_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                productCode = "P060100206",
                productName = "사잇돌",
                active = true
            ),
            Product.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.LINE_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                productCode = "P060100205",
                productName = "드림론",
                active = true
            ),
            Product.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.KAKAO_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                productCode = "TA",
                productName = "갈아타기OK론",
                active = true
            ),
            Product.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.TOSS_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                productCode = "FNQ005",
                productName = "kiwi비상금",
                active = true
            ),
            Product.create(
                partner = initialPartner.firstOrNull { it.partnerCode == PartnerCode.SHINHAN_BANK },
                loanType = LoanType.PERSONAL_CREDIT,
                productCode = "0201001074",
                productName = "비상금신한론",
                active = true
            )
        )
        productRepository.saveAll(initialProduct)

        val initialUser = listOf(
            Member().apply {
                name = "윤교희"
                mobile = "01056677055"
                email = "gyohee91@gmail.com"
            }
        )

        memberRepository.saveAll(initialUser)
    }
}
